build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalTestArtifactBasePlugin.java

```diff
@@ -25,6 +26,7 @@ public InternalTestArtifactBasePlugin(ProviderFactory providerFactory) {
 
     @Override
     public void apply(Project project) {
+        project.getPlugins().apply(JavaPlugin.class);
```

Comment:
`javaPluginExtension.registerFeature` that is done in `InternalTestArtifactExtension` requires now the java plugin to be applied. this is why we add the java plugin here.